libs/wingc/src/lib.rs

```diff
@@ -273,6 +275,12 @@ pub fn compile(source_path: &Path, out_dir: Option<&Path>) -> Result<CompilerOut
 		Diagnostics::new()
 	};
 
+	// Validate that every Expr has an evaluated_type
+	if cfg!(debug_assertions) {
```

Comment:
I hope this is always on

Reply:
Right now it's only on during dev (local and CI builds). Should we also surface these in our release build?

Reply:
I'd add to release builds too